include/psa/crypto_values.h

```diff
@@ -414,6 +414,20 @@
  */
 #define PSA_KEY_TYPE_DERIVE                         ((psa_key_type_t)0x1200)
 
+/** A low-entropy secret for password hashing or key derivation.
+ *
+ * The key policy determines which key derivation algorithm the key
+ * can be used for.
+ */
+#define PSA_KEY_TYPE_PASSWORD                       ((psa_key_type_t)0x1300)
+
+/** A secret value that can be mixed in when doing password hashing.
```

Comment:
(Using this thread to reply to @athoelke)

> How valuable is the provision of a distinct key type for pepper, rather than reusing RAW_DATA? Or is pepper a shared secret, which warrants a distinction from RAW_DATA?

A pepper is a share secret.

`RAW_DATA` is intended for data that has no direct cryptographic relevance or structure, at least as far as the PSA API is concerned. (A common use is to store a certificate alongside a private key. This has cryptographic relevance, but the crypto API doesn't use the certificate object, which is why it hasn't gotten its own type, though it might in a future version.) Even for a salt, using `RAW_DATA` is pushing it: a salt does have cryptographic “structure” in that it's meant to be unique, and it definitely has cryptographic relevance as the input to a cryptographic algorithm.